src/d3a/d3a_core/simulation.py

```diff
@@ -371,16 +371,10 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
                         external_global_statistics.update()
 
                 self.area.tick_and_dispatch()
+                bid_offer_matcher.event_tick(
+                    is_it_time_for_external_tick=external_global_statistics.
+                    is_it_time_for_external_tick(current_tick_in_slot))
```

Comment:
So you need to pass `is_it_time_for_external_tick` ? `external_global_statistics` is a singleton and you could import it in MycoMatcher right? (or is there a circular import again ?)

Reply:
Right, there was a circular import error, I could import it locally but still I will need to pass the `current_tick_in_slot` anw, so I believe passing the flag makes more sense

Reply:
Regarding circular imports: this might be another reason to replace the variable in `singletons.py` with an actual singleton class. We could put it in another module (since `singletons.py` doesn't give information about the type of objects it contains, besides them generically being singletons). At that point, we could just have something like:
```python
from d3a.models.myco_matcher.myco_matcher_singleton import MycoMatcherSingleton
matcher = MycoMatcherSingleton()
```
This would create a new `MycoMatcher` when it doesn't already exist, and it will return the existing one if it exists (singleton).

Reply:
@fievelk this is basically the same, the `simulation.py` would still import both `MycoMatcher` and `external_global_statistics`, the `MycoMatcher` will still import `ExternalMatcher` that in its turn will import the `external_global_statistics` and we fall back in the same maze, however as I stated we would have to pass an argument in any case (current_tick_in_slot), so why not passing the flag` is_time_for_external_tick` instead and save imports?

Reply:
I don't think that the situation you described would generate a circular import. If you have the time you could give it a try.

Reply:
Okay I'll give it a try but won't design a singleton Myco matcher, I'll just move the instantiation to another module as I believe both will have the same impact

Reply:
If this succeeded, I'll either make it singleton or permanently move the instantiation somewhere else

Reply:
> Right, there was a circular import error, I could import it locally but still I will need to pass the `current_tick_in_slot` anw, so I believe passing the flag makes more sense

Indeed, we need to pass this anyway. Didn't think about it. So. If you don't succeed with removing the circular import, just keep as you proposed, IMO

Reply:
It works!
Great, so we have two options here

1. From the `simulations.py` pass the `current_tick_in_slot` and in the external matcher import `external_global_statistics` and check if it is time for external tick (Move the Myco matcher singleton as in the suggestion you provided)
2. Directly pass the is_time_for_external_tick from `simulation.py` to external matcher (Leave the currentsingletons)

Reply:
Both are fine for it. Since the current tick must unfortunately be passed in any case, I think this wouldn't make a big difference. Maybe the 2nd solution would be lighter, but do what you think is best!